pynicotine/plugins/autobahn/__init__.py

```diff
@@ -0,0 +1,248 @@
+                progressbar_fraction = self.frame.userbrowse.pages[user].progress_bar.get_fraction()
+                
+                self.log("total %s, folders %s, progress %s, info_text %s" % (total_size_text, total_folders_text, progressbar_fraction, info_box_text))
+                
+                refreshing = True
+
+                total_size_array = total_size_text.split(" ")
+                total_size_number = float(total_size_array[0])
+                total_size_unit = total_size_array[1]
+
+                if (progressbar_fraction == 1):
+                    self.log("fraction true");
+
+                if ("empty" in info_box_text):
+                    self.log("empty infobox true");
+                    
+                if (total_size_number > 0):
+                    self.log("size >0 true");
+
+                if (float(total_folders_text) > 0):
+                    self.log("folders >0 true");
+
+                if ((progressbar_fraction == 1) and ("offline" in info_box_text)):
+                    refreshing = False
+
+                if ((progressbar_fraction == 1) and ("empty" in info_box_text)):
+                    refreshing = False
+                
+#                if ((progressbar_fraction == 1) and ((total_size_text != "0.0 B") or (total_folders_text != "0"))):
+                if ((progressbar_fraction == 1) and ((total_size_number > 0) or (float(total_folders_text) > 0))):
+                    refreshing = False
+
+            except KeyError:
+                return
+
+            
+            if refreshing:
+                self.retries[user] = self.retries[user] + 1
+                #user browse sometimes gets stuck - so every 100 tries (roughly 5-10s) new request will be made to push it
+                if (self.retries[user] % 100 == 0):
+                    self.log("Still waiting for files from user %s, retry %s" % (user, self.retries[user]))
+                    self.core.userbrowse.browse_user(user)
+                self.core.queue.append(slskmessages.GetUserStats(user))
+                return
+
+            #if cannot connect message appeared - stop checking
+            info_box_text = self.frame.userbrowse.pages[user].info_bar.label.get_text()
+            if ("offline" in info_box_text):
+                return
+
+            total_size_text = self.frame.userbrowse.pages[user].share_size_label.get_text()
+            total_folders_text = self.frame.userbrowse.pages[user].num_folders_label.get_text()
+
+            total_size_array = total_size_text.split(" ")
+            total_size_number = float(total_size_array[0])
+            total_size_unit = total_size_array[1]
+            #self.log("after split %s, %s" % (total_size_number, total_size_unit))
+            if (total_size_unit == "B"):
+                total_size = total_size_number
+            elif (total_size_unit == "KiB"):
+                total_size = total_size_number * 1024
```

Comment:
You could use the `factorize()` function which is provided in the `utils` module:

https://github.com/nicotine-plus/nicotine-plus/blob/8a083b78201c4dce19be4279b4c6e05b7e4d17a7/pynicotine/utils.py#L136-L172

But it would be needed to get the raw size from the `core` anyway, such conversion is intended only for user inputs.